railties/test/railties/engine_test.rb

```diff
@@ -1511,21 +1537,25 @@ def app
     end
 
     # Restrict frameworks to load in order to avoid engine frameworks affect tests.
-    def restrict_frameworks
+    def restrict_frameworks(remove: [])
```

Comment:
_I started with the assumption that removing ActiveJob during the "boot" process would be the appropriate approach, but it was not enough as I noted on lines 906-908. As such, it's possible this part of the change is no longer necessary._

Reply:
This might be tricky because we boot rails per test. If we can't emulate the `NoMethodError` we could try mocking it instead.

Reply:
@gmcgibbon - I'm not super familiar with minitest anymore, and it looks like stubbing this out correctly is just not possible without bringing in another test double library, which doesn't makes sense. (I don't want to get into an argument about libraries, but in RSpec stubbing this out without affecting the rest of `config` would be straightforward.)

Reply:
😄 Yes, rspec is much better at stubbing/mocking than minitest, I agree. Thinking further I think you can just forgo mocking and do this:

```
test "seed data can be loaded when ActiveJob is not present" do
  @plugin.write "db/seeds.rb", <<-RUBY
    Bukkits::Engine.config.bukkits_seeds_loaded = true
  RUBY

  app_file "db/seeds.rb", <<-RUBY
    Rails.application.config.app_seeds_loaded = true
  RUBY

  boot_rails

  # In a real app, active_job would be undefined
  undefine_config_option(:active_job)

  Rails.application.load_seed
  assert Rails.application.config.app_seeds_loaded
  Bukkits::Engine.load_seed
  assert Bukkits::Engine.config.bukkits_seeds_loaded
end

private

  def undefine_config_option(name)
    Rails.application.config.class.class_variable_get(:@@options).delete(name)
  end
```

Reply:
@gmcgibbon - that's great! I've added both of your suggestions, and will squash the commits and force push my branch after this is approved.